### DISCUSSION TITLE
Recording Session

### DISCUSSION THREAD

**Discussion by AswinParanji** [neutrinolabs/xrdp#2215] [Ideas]
Hi,
Is there any way to record a session using FFMPEG? If that is not available, could you point me where exactly I'll get bitmap that can be sent to ffmpeg for h264 recording?

Regards
Aswin Paranji

**Comment by matt335672** [comment]
That's not something we support at the moment.

Are you in a position to do the recording at the client? That gives you a large range of options to choose from.

**Reply by AswinParanji** [reply]
Dear Matt,
Thanks for responding to the question I posted.
What I'm doing is I'm using xrdp and neutrinordp to proxy a target server.
My setup is like this.
Laptop->NeutrinoRDP/xrdp server->target server

I need to record the RDP data coming from the target server in NeutrinoRDP/xrdp server not in the Laptop.
If you at least tell me which file I need to look into to add ffmpeg code to record, that would be of great help.

Best Regards,
Aswin Paranji

**Comment by matt335672** [comment]
xrdp itself doesn't contain a complete copy of the framebuffer (it doesn't need one), so you haven't got enough information available there to do what you want.

It's possible there's something useful in NeutrinoRDP, but I'm afraid I don't know that part of the code well enough to advise. It's based on FreRDP, and that don't seem to offer what you want either, so I suspect this isn't a useful way forward either.

In any case, doing the compression without compromising performance will be quite tricky.

I still reckon you're better targeting one of the endpoints from a technical perspective.

Technical considerations aside, adding a facility like this to a protocol which can be MitMed (if you aren't using TLS, anyway) is questionable from a moral perspective.

**Comment by fuqifacai** [comment]
> Hi, Is there any way to record a session using FFMPEG? If that is not available, could you point me where exactly I'll get bitmap that can be sent to ffmpeg for h264 recording?
> 
> Regards Aswin Paranji

You could use any solution to capture the GUI of xrdp rendered , then save to bitmap. 
24 bitmap /per second to make a video by using FFMPEG library.
This is the idea.
If you need more detailed codes, please contact me directly, of course I think you could also get it via google .
